packages/gateway/src/gateway.js

```diff
@@ -58,6 +68,8 @@ export async function gatewayGet(request, env, ctx) {
         await Promise.all([
           storeWinnerGwResponse(request, env, winnerGwResponse),
           settleGatewayRequests(),
+          // Cache request URL in Cloudflare CDN
+          cache.put(request.url, winnerGwResponse.response.clone()),
```

Comment:
We can only cache things < 512MB. We should check `Content-Length` before attempting.

This just occurred to me - it would be really great to know the distribution of file sizes the gateway is serving. Can we please add file size to the metrics as a histogram? (separate PR obvs)

Reply:
Good idea! I will do the validation and track metrics in follow up PR